### PR TITLE
Use http/https

### PR DESCRIPTION
**Pull request by glusa8**
Reduce dependency on `superagent` by replacing it w/  `http`/`https` library

### CODE REVIEW

File: src/airship.js [hunk 1]
```diff
@@ -102,13 +104,10 @@ export default class Airship extends Environment {
             version: version
           }
         })
-        .then(res => {
-          if (!res.ok) {
-            logger('Something went wrong. Ingestion failed')
-          }
-        })
+      )
+        .then(result => {})
```
Comment (glusa8) [inline src/airship.js]: I think this will just be the result if any. There is no longer the status.

File: src/airship.js [hunk 2]
```diff
@@ -102,13 +103,10 @@ export default class Airship extends Environment {
             version: version
           }
         })
-        .then(res => {
-          if (!res.ok) {
-            logger('Something went wrong. Ingestion failed')
-          }
-        })
+      )
+        .then(() => {})
```
Comment (bsuh) [inline src/airship.js]: unnecessary line

File: src/airship.js [hunk 3]
```diff
@@ -162,16 +158,102 @@ export default class Airship extends Environment {
...
+  getContent(url, timeout = REQUEST_TIMEOUT) {
+    return new Promise((resolve, reject) => {
+      const urlObj = URL.parse(url)
+
+      const lib = urlObj.protocol === 'https:' ? https : http
+
+      const request = lib.get(url, response => {
+        if (response.statusCode < 200 || response.statusCode > 299) {
+          reject(
+            new Error(
```
Comment (bsuh) [inline src/airship.js]: are we rejecting with error or string? it's inconsistent

Comment (bsuh) [reply src/airship.js]: https://github.com/airshiphq/flagger-js/blob/master/docs/api-reference.md#flaggerseterrorlistenerlistener suggests that we should only logger strings

**Pull request merged**